op-challenger/flags/flags.go

```diff
@@ -379,6 +383,39 @@ func getL2Rpc(ctx *cli.Context, logger log.Logger) (string, error) {
 	return l2Rpc, nil
 }
 
+func FactoryAddress(ctx *cli.Context) (common.Address, error) {
+	if ctx.IsSet(FactoryAddressFlag.Name) && ctx.IsSet(flags.NetworkFlagName) {
+		return common.Address{}, fmt.Errorf("flag %v and %v must not both be set", FactoryAddressFlag.Name, flags.NetworkFlagName)
+	}
+	if ctx.IsSet(FactoryAddressFlag.Name) {
+		gameFactoryAddress, err := opservice.ParseAddress(ctx.String(FactoryAddressFlag.Name))
+		if err != nil {
+			return common.Address{}, err
+		}
+		return gameFactoryAddress, nil
+	}
+	if ctx.IsSet(flags.NetworkFlagName) {
+		chainName := ctx.String(flags.NetworkFlagName)
+		chainCfg := chaincfg.ChainByName(chainName)
+		if chainCfg == nil {
+			var opts []string
+			for _, cfg := range superchain.OPChains {
+				opts = append(opts, cfg.Chain+"-"+cfg.Superchain)
+			}
+			return common.Address{}, fmt.Errorf("unknown chain: %v (Valid options: %v)", chainName, strings.Join(opts, ", "))
+		}
+		addrs, ok := superchain.Addresses[chainCfg.ChainID]
+		if !ok {
+			return common.Address{}, fmt.Errorf("no addresses available for chain %v", chainName)
+		}
+		if addrs.DisputeGameFactoryProxy == (superchain.Address{}) {
```

Comment:
We will need to follow up and add the mainnet DGF proxy to the superchain registry.

Reply:
Yep definitely.  I was contemplating allowing use of `--network` and `--dispute-game-factory` so you can override the factory address.  Might make that change since it will then work better with op-mainnet and other chains that are setting up infra prior to actually triggering the portal upgrade and updating the super chain registry.  Plus lets us more easily test against a different deployment on a network if we want to do that without actually respecting it for withdrawals yet.